Source/WebCore/platform/ttsclient/PlatformSpeechSynthesizerTTSClient.cpp

```diff
@@ -0,0 +1,409 @@
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDERS. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "PlatformSpeechSynthesizerTTSClient.h"
+
+#if ENABLE(SPEECH_SYNTHESIS) && USE(TTS_CLIENT)
+#include "Page.h"
+#include "PlatformSpeechSynthesisUtterance.h"
+
+#include <sys/types.h>
+#include <unistd.h>
+
+#define MAX_ALLOWED_TEXT_LENGTH 1024
+
+#define CHECK_TTS_SESSION(utterance)                                                                                                                \
+    do {                                                                                                                                            \
+        if (!m_ttsConnected) {                                                                                                                      \
+            notifyClient(utterance, SpeechSynthesisErrorCode::SynthesisUnavailable);                                                                \
+            return;                                                                                                                                 \
+        }                                                                                                                                           \
+        if (!m_ttsClient || !m_ttsSessionId) {                                                                                                      \
+            notifyClient(utterance, SpeechSynthesisErrorCode::NotAllowed);                                                                          \
+            return;                                                                                                                                 \
+        }                                                                                                                                           \
+    } while (0)
+
+namespace {
+static bool bSpeechSynthOverrideSysTTSConfig = getenv("SPEECH_SYNTHESIS_OVERRIDE_SYSTEM_TTS_CONFIG");
+} // namespace
+
+namespace WebCore {
+
+// PlatformSpeechSynthesizer
+WEBCORE_EXPORT Ref<PlatformSpeechSynthesizer> PlatformSpeechSynthesizer::create(PlatformSpeechSynthesizerClient& client)
+{
+    return adoptRef(*new PlatformSpeechSynthesizerTTSClient(client));
```

Comment:
This unconventional... Can you make a TTS wrapper, implement stuff there and then call the wrapper from the PlatformSpeechSynthesizer class?

Reply:
We use this approach for some time, flite wasn't there (nor glib version of speech sythesis as in 2.38). There is no need to make any changes in PlatformSpeechSynthesizer.h/cpp with such approach, on adding new implementation. But you are right that it doesn't match the rest...
If you prefer wrapper approach then it's fine with me

Reply:
You'd need changes only in the header, not the cpp file, afaict.

Reply:
Fixed, converted into wrapper design similar to other impls